### ISSUE TITLE
[iOS] Uncaught Kotlin exception: kotlin.IllegalStateException: Internal kotlinx.rpc error: unable to find class UserService rpc client object - on Sampled app

### ISSUE THREAD

**Issue by jerryOkafor** [Kotlin/kotlinx-rpc#93]
I tried following the Readme guidelines and ended up with the following error:

```
Uncaught Kotlin exception: kotlin.IllegalStateException: Internal kotlinx.rpc error: unable to find class UserService rpc client object
```

To be clear, Android works as expected and to be sure I am not making any mistakes, I pulled the sample, tried it on iOS, and ended up with the same issue.

Environment:
Kotlin: 1.9.24
Xcode Version : 15.4 (15F31d)

**Comment by Mr3zee** [comment]
Thanks for the report! I will investigate

**Comment by afTrolle** [comment]
I'm also having the similar issue, However I have a different project setup than the sample.

I moved the RPC interface to it's own module that then the server and client modules depend on.

`kotlin.IllegalStateException: Internal kotlinx.rpc error: unable to find class run.tevla.api.ForgeServiceContract rpc client object`

![image](https://github.com/Kotlin/kotlinx-rpc/assets/1321086/d85054b3-bef7-44af-b19d-6d3ec411369a)

**Comment by jerryOkafor** [comment]
This is similar to how my modules are arranged. When I noticed the issue, I pulled the sample and ran it to verify that the same issue existed.

**Comment by ShayOinif** [comment]
Hello.
I am facing the same issue and would love to have a fix for it.
Thanks!

**Comment by ShayOinif** [comment]
Seems like the problem is present on macOS targets as well.
Thought maybe I misconfigured my gradles, but in the sample from the code base the issue reoccurred.

**Comment by ShayOinif** [comment]
Tested and found the same issue on Linux targets as well.

**Comment by xxfast** [comment]
Only seems to be affecting the native targets - JVM and Android targets works as expected 👍

**Comment by Mr3zee** [comment]
Fixed in #112
